ats/pql/pql.py

```diff
@@ -0,0 +1,186 @@
+
+        searchingVariable = current_token
+        current_token = get_next_token()
+
+        return searchingVariable
+
+    def match_name_token():
+        assert_token("NAME_TOKEN")
+        nonlocal current_token
+        if not is_name_token(current_token):
+            raise ValueError(f"Token '{current_token}' is not a valid NAME_TOKEN")
+
+        name = current_token
+        current_token = get_next_token()
+
+        return name
+
+    def match_variable_in_query_token(variables):
+        assert_token("VARIABLE_IN_QUERY_TOKEN")
+        nonlocal current_token
+        # w przyszłości jak zostanie dodane "and" do pql trzeba będzie to rozszerzyć o ilość zmiennych występujących w zapytaniu, 
+        # póki co są tylko 2 więc narazie to wystarczy
+        if current_token != variables[0] and current_token != variables[1]:
+            raise ValueError(f"Token '{current_token}' is not a valid VARIABLE_IN_QUERY_TOKEN")
+
+        current_token = get_next_token()
+
+
+    def match_design_entity_relationship():
+        assert_token("DESIGN_ENTITY_RELATIONSHIP_TOKEN")
+        nonlocal current_token
+        if not is_program_design_entity_relationship_token(current_token):
+            raise ValueError(f"Token '{current_token}' is not a valid NAME_TOKEN")
+
+        relationship = current_token
+        current_token = get_next_token()
+        if current_token == "*": 
+            relationship += "*"
+            current_token = get_next_token()
+
+        return relationship
+
+
+    def match_end_of_declaration_token():
+        assert_token("NAME_DECLARATION_TOKEN")
+        nonlocal current_token
+        if current_token != "," and current_token != ";":
+            raise ValueError(f"Token '{current_token}' is not a valid NAME_DECLARATION_TOKEN")
+
+        name = current_token
+        current_token = get_next_token()
+
+        return name
+
+    def match_parameter_token(variables):
+        assert_token("RELATIONSHIP_PARAMETER_TOKEN")
+        nonlocal current_token
+        if not is_string_token(current_token) and not is_integer_token(current_token) and variables[current_token] is None:
+            raise ValueError(f"Token '{current_token}' is not a valid RELATIONSHIP_PARAMETER_TOKEN")
+            
+        parameter = current_token
+        current_token = get_next_token()
+
+        return parameter
+
+    def process_pql_code():
+        nonlocal current_token
+        current_token = get_next_token()
+        variables = process_variable({})
+        process_operation(variables)       
+
+        return None
+
+    def process_operation(variables):
+        nonlocal current_token
+        while is_variable_type_token(current_token):
+            variables = process_variable(variables)
+
+        process_query(variables)
+
+        return None
+
+    def process_query(variables):
+        nonlocal current_token
+        match_token("Select")
```

Comment:
Ogolem w PQL mozesz dawac wiele zapytan. Zapytanie konczy sie srednikiem. Jak dostaniesz srednik, to dopiero wykonujesz zapytania do API design extractora i zwracasz wynik.
```
stmt s1, s2;
Select s1 such that Follows(s1, s2);
Select s1 such that Follows(s2, s1);
```

Reply:
![image](https://user-images.githubusercontent.com/65611113/229427891-932b0238-b57c-42bb-8891-07465aa963f4.png)

W pliku z iteracją 1 nie ma żadnych średników poza deklaracją zmiennych, dlatego nie zostało to zaimplementowane

Reply:
no to jak masz kolejne wyrazenie / koniec pliku to wtedy wykonujesz poprzednie